src/Components/Web.JS/src/Boot.WebAssembly.ts

```diff
@@ -122,14 +129,16 @@ async function boot(options?: Partial<WebAssemblyStartOptions>): Promise<void> {
 
   const bootConfigResult: BootConfigResult = await bootConfigPromise;
   const jsInitializer = await fetchAndInvokeInitializers(bootConfigResult.bootConfig, candidateOptions);
+  const progressService = new WebAssemblyProgressService();
+  progressService.attach(Blazor.webAssemblyLoadingSetProgress!);
```

Comment:
Hmm just realized in the case when `document.getElementById('blazor-default-loading') == undefined` (ie no loading DOM elements), and `Blazor.webAssemblyLoadingSetProgress == undefined` (no user defined function + no loading DOM elements so no default function), this'll actually be `undefined`, so the `!` may not be a safe assumption.

```suggestion
  if (Blazor.webAssemblyLoadingSetProgress) {
    progressService.attach(Blazor.webAssemblyLoadingSetProgress);
  }
```

Can you please try out the case when you delete all the HTML DOM elements related to the loading stuff (ie revert the index.html to what we have currently), and confirm everything is still functioning normally otherwise? We want to make sure our updates to the `blazor.webassembly.js` don't break existing apps.

Also, if you haven't already, can you please make a makeshift `setProgress` function to simulate the user creating their own function to see if that scenario is working as expected?

Reply:
Everything still functions normally after deleting all of the new HTML DOM elements and reverting to the previous loading screen. 

Adding a custom `setProgress` function in index.html is working as expected too and displays the correct number of resources. Everything is working correctly with my custom implementation of a loading bar.